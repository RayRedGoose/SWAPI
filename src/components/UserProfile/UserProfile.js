import React from 'react';
import './UserProfile.scss'
import userImage from '../../images/death-star.svg'

const UserProfile = ({name, rank, quote, logOut}) => {

  return (
    <section className="user-profile">
      <img src={userImage}></img>
      <section>
        <h3>{name}:<span> {rank}</span></h3>
        <p>{quote}</p>
      </section>
      <button onClick={logOut}>Log Out</button>
    </section>
  )
}

export default UserProfile;
