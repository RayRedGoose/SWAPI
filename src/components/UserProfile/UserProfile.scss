.user-profile {
  display: flex;
  position: relative;
  top: 0; right: 0;
  div {
    width: 100px; height: 100px;
    background-color: #F1AA61;
    background-size: 80%;
    background-repeat: no-repeat;
    background-position: center;
    margin-right: 20px;
  }
  section {
    min-width: 200px;
    max-width: 250px;
    position: relative;
    * {
      margin: 0; padding: 0;
    }
    h3 {
      color: #000;
      font-size: 24px;
      text-transform: uppercase;
      span {
        color: #5A2C05;
        text-transform: lowercase;
        font-style: italic;
        font-weight: 400;
        margin-left: 10px;
      }
    }
  }
  p {
    margin-top: 15px;
    margin-bottom: 40px;
    &:before, &:after {
      content: '"'
    }
  }
  button {
    cursor: pointer;
    background: none;
    border: none;
    color: rgba(90, 44, 5, 0.5);
    font-size: 18px;
    position: absolute;
    right: 0; bottom: 0;
    z-index: 30;
    padding: 0.4em;
    &:hover {
      color: rgba(90, 44, 5, 1);
    }
  }
}

.user-profile {
  &:before {
    content: '';
    width: 140px; height: 70px;
    position: absolute;
    top: 0; left: 0;
    margin: -20px;
    border-top: 2px solid rgba(90, 44, 5, 1);
    border-left: 2px solid rgba(90, 44, 5, 1);
  }

  &:after {
    content: '';
    width: 140px; height: 70px;
    position: absolute;
    bottom: 0; right: 0;
    margin: -20px;
    border-bottom: 2px solid rgba(90, 44, 5, 1);
    border-right: 2px solid rgba(90, 44, 5, 1);
  }
}
