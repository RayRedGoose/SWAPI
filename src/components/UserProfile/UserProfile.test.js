import React from 'react';
import { shallow } from 'enzyme';
import UserProfile from './UserProfile';


describe("UserProfile", () => {

  it("should render UserProfile", () => {
    const wrapper = shallow(
      <UserProfile
      user={{name: "The Rock"}}
      logOut={jest.fn()}
      />
    )
    expect(wrapper).toMatchSnapshot();
  })

})
