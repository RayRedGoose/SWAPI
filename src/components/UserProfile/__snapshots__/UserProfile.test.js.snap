// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should match snapshot if toLoginPage state is true 1`] = `
<Redirect
  to="/"
/>
`;

exports[`UserProfile should render UserProfile 1`] = `
<section
  className="user-profile"
>
  <div
    style={
      Object {
        "backgroundImage": "url(death-star.svg)",
      }
    }
  />
  <section>
    <h3>
      The Rock
      <span>
         
        expert
      </span>
    </h3>
    <p>
      I hate sand
    </p>
    <button
      onClick={[Function]}
    >
      Log Out
    </button>
  </section>
</section>
`;
